File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }
        int[] charCountS = new int[26];
        int[] charCountT = new int[26];
        for (char c : s.toCharArray()) {
            charCountS[c - 'a']++;
        }
        for (char c : t.toCharArray()) {
            charCountT[c - 'a']++;
        }
        return Arrays.equals(charCountS, charCountT);
    }
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        s_freq = defaultdict(int)
        for i in s:
            s_freq[i] += 1
        for i in t:
            if i not in s_freq or s_freq[i]==0:
                return False
            else:
                s_freq[i]-= 1
        return True
                
        